### PR TITLE
feat: add knapsack renderer

### PR DESCRIPTION
**Pull request by EvanLovely**
Recreates #1605 

## Changes

- Includes this PR and likely makes that PR not needed as some requested changes have been made on this branch: https://github.com/johnsonandjohnson/Bodiless-JS/pull/1605
- Adds Knapsack site to `sites/knapsack`
- Adds the `cardSpec` export to `packages/card` - however there have been type changes to `TokenCollection` which is used in `KnapsackBodilessSpec` (which is what `cardSpec` is) that are failing. I'll need Chris Oden to make any updates to this as we had it working when we were using `@bodiless/fclasses@ks-canary`
- The package `@bodiless/knapsack-renderer` version has been changed from `3.33.4` to `1.0.0-beta.7` in order to line up with rest of repo's package versions.

## Test Instructions

```bash
cd sites/knapsack
npm start
```

### CODE REVIEW

File: packages/knapsack-renderer/src/renderer-bodiless.ts [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+import { getDemoAppUsage, getUsage } from './utils';
+import { KnapsackBodilessSpec } from './types';
+
+const { pkg } = findUpPkgJson(__dirname);
+
+const ourImports: KsJsImport[] = [
+  {
+    type: 'extra',
+    importInfo: {
+      path: '@canvasx/elements',
```
Comment (VancheeZze) [inline packages/knapsack-renderer/src/renderer-bodiless.ts:28]: Wonder if any of the Bodiless packages should refer to any of the `@canvasx` packages. cc @wodenx

**Review by VancheeZze — CHANGES_REQUESTED** [review]
Awesome work! Looks good! Just a few minor commennts

File: packages/knapsack-renderer/package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "@bodiless/knapsack-renderer",
+  "description": "Render Bodiless templates in Knapsack",
+  "version": "3.33.4",
```
[the remark this answers is not in the document]
Comment (EvanLovely) [reply packages/knapsack-renderer/package.json:4]: Version has been updated to be in-line

File: packages/knapsack-renderer/package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    "@knapsack/typescript-config-starter": "^3.47.0",
+    "@types/babel__core": "^7.1.16",
+    "@types/node": "^16",
+    "@types/react": "^17.0.39",
+    "@types/react-dom": "^17.0.11",
+    "ava": "^3.15.0",
+    "ts-expect": "^1.3.0",
+    "typescript": "^4.0.0"
+  },
+  "license": "GPL-2.0-or-later",
```
[the remark this answers is not in the document]
Comment (EvanLovely) [reply packages/knapsack-renderer/package.json:48]: Changed the license

File: packages/knapsack-renderer/readme.md
[the remark this answers is not in the document]
Comment (EvanLovely) [reply packages/knapsack-renderer/readme.md:1]: I am not sure how the `bodiless.docs.json` work - feel free to add it to the package to get your docs site building like you'd like. I did rename `readme.md` to `README.md` and added the basic info you asked for though.

**Comment by EvanLovely** [comment]
- Since the work is now in @wodenx's [fork](https://github.com/wodenx/Bodiless-JS) there has been a new PR made: #2008

**Comment by EvanLovely** [comment]
- Superseded by #2008

**Pull request closed without merging**